Core/Src/RTD.c

```diff
@@ -0,0 +1,54 @@
+/*
+ * RTD_handler.c
+ *
+ *  Created on: Aug 15, 2021
+ *      Author: Felipe Telles
+ */
+
+#include "RTD.h"
+#include "global_variables.h"
+#include "debugleds.h"
+#include "main.h"
+#include "global_instances.h"
+
+void RTD_handler(void *argument) {
+
+    for(;;) {
+
+        #ifdef DEBUG_ECU
+        extern void brkpt();
+        brkpt();
+        #endif
+
+        osSemaphoreRelease(s_allowed_change_modeHandle);                                //libera semáforo que permite a mudança de modos
```

Comment:
muda isso pra flag de thread, igual na linha 26

Reply:
> muda isso pra flag de thread, igual na linha 26

Precisaria usar a função de limpar a flag, fora da thread de modos (na de RTD), o que não é permitido. Poderia usar event flag para isso, mas prefiro a lógica com semáforo
[The function osThreadFlagsClear clears the specified flags for the currently running thread.](https://www.keil.com/pack/doc/CMSIS/RTOS2/html/group__CMSIS__RTOS__ThreadFlagsMgmt.html#ga656abc1c862c5b9a2b13584c42cc0bfa)

Mudei a lógica do botão para usar ThreadFlags.